packages/core/src/axis-chart.ts

```diff
@@ -63,61 +45,109 @@ export class AxisChart extends Chart {
 
 		// Decide the position of the legend in reference to the chart
 		let fullFrameComponentDirection = LayoutDirection.COLUMN;
-		const legendPosition = Tools.getProperty(this.model.getOptions(), "legend", "position");
-		if (legendPosition === LegendPositions.LEFT) {
-			fullFrameComponentDirection = LayoutDirection.ROW;
 
-			if (!this.model.getOptions().legend.orientation) {
-				this.model.getOptions().legend.orientation = LegendOrientations.VERTICAL;
-			}
-		} else if (legendPosition === LegendPositions.RIGHT) {
-			fullFrameComponentDirection = LayoutDirection.ROW_REVERSE;
+		// Check if the legend is enabled for the chart
+		if (this.model.getOptions().legend.enabled) {
```

Comment:
What happens if I disable legend in the initial render, then want to enable again?